src/App.tsx

```diff
@@ -17,18 +50,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                setFilter={setFilter}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!todos.length ? (
+                <Loader />
+              ) : (
```

Comment:
You should show the loader only if you are loading something. If the API returns an empty array the user sees an infinite loader